src/lib/mergeDefault.ts

```diff
@@ -3,21 +3,21 @@ import { isObject } from './isObject';
 
 import type { DeepRequired } from './utilityTypes';
 
-type KeyedObject = Record<PropertyKey, unknown>;
+type NonNullObject = {};
 
 /**
  * Sets default properties on an object that aren't already specified.
  * @since 0.5.0
  * @param def Default properties
  * @param given Object to assign defaults to
  */
-export function mergeDefault<A extends KeyedObject, B extends Partial<A>>(defaults: A, given?: B): DeepRequired<A & B> {
+export function mergeDefault<A extends NonNullObject, B extends Partial<A>>(defaults: A, given?: B): DeepRequired<A & B> {
 	if (!given) return deepClone(defaults) as DeepRequired<A & B>;
 	for (const key in defaults) {
```

Comment:
While you're at it, why are we using for-in?

Reply:
Resolved in 101b66d.

Reply:
Come again? you're still using `in`